gangplank/ocp/ssh.go

```diff
@@ -0,0 +1,139 @@
+package ocp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/coreos/gangplank/spec"
+	log "github.com/sirupsen/logrus"
+)
+
+type SSHForwardPort struct {
+	Host string
+	User string
+	Pass string
+
+	// port is not exported
+	port int
+}
+
+// getSshMinioForwarder returns an SSHForwardPort from the jobspec
+// definition for forwarding a minio server, or nil if forwarding is
+// not enabled.
+func getSshMinioForwarder(j *spec.JobSpec) *SSHForwardPort {
+	if j.Job.MinioSSHForward == "" {
+		return nil
+	}
+	return &SSHForwardPort{
+		Host: j.Job.MinioSSHForward,
+		User: j.Job.MinioSSHUser,
+		Pass: j.Job.MinioSSHPass,
+	}
+}
+
+// sshForwarder is a generic forwarder from the local host to a remote host
+func sshForwarder(ctx context.Context, cfg *SSHForwardPort) (chan<- bool, error) {
+	termCh := make(chan bool, 2048)
+
+	if cfg == nil {
+		return nil, errors.New("configuration for SSH forwrding is nil")
+	}
+
+	var cmd *exec.Cmd
+	run := func() error {
+		host := cfg.Host
+		if cfg.User != "" && !strings.Contains(cfg.User, "@") {
+			host = fmt.Sprintf("%s@%s", cfg.User, cfg.Host)
+		}
+		args := []string{"ssh", "-o", "ServerAliveInterval=15", "-o", "ServerAliveCountMax=5", "-N", "-R",
+			fmt.Sprintf("%d:127.0.0.1:%d", cfg.port, cfg.port), host,
+		}
```

Comment:
I think this still needs to be addressed

Reply:
we also need to add plumbing for getting the identity file (keyfile) to this command. Podman takes care of it because it's in the CONTAINER_SSHKEY env var, but we need to tack it into here (similar to how password was done before).

Reply:
I've been testing using the ssh agent, so that path works nicely for now.